### PR TITLE
Voting: Add full-coverage unit tests

### LINKED ISSUE #84: Write full-coverage unit tests for CRVoting
Please follow the convention used for the `JurorsRegistry` tests, some of them are:
- Avoid mocking unnecessarily, only when there is no other option
- Use `test-helpers` module provided by `@aragon`
- Implement reusable helpers stuff instead of custom non-exported functions
- Use `let` scoped variables instead of properties to define test variables

### PR DESCRIPTION
**Pull request by facuspagnuolo**
Fixes #84 

This PR includes unit tests for
- [x] Initialization
- [x] General voting queries
- [x] Create votings
- [x] Commit votes
- [x] Leak votes
- [x] Reveal votes
- [x] Integration tests

### CODE REVIEW

File: test/voting/crvoting-integration.js [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+      const winningOutcome = await voting.getWinningOutcome(votingId)
+      assert.equal(winningOutcome.toString(), OUTCOMES.MISSING, 'winning outcome should be missing')
+
+      const winningOutcomeTally = await voting.getWinningOutcomeTally(votingId)
+      assert.equal(winningOutcomeTally.toString(), 0, 'winning outcome tally should be zero')
+    })
+  }
+
+  describe('integration', () => {
+    context('when non of the voters committed a vote', () => {
```
Comment (izqui) [inline test/voting/crvoting-integration.js]:
```suggestion
    context('when none of the voters committed a vote', () => {
```

File: test/voting/crvoting-integration.js [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+        [voterWeighted1]:  { weight: 1,  outcome: undefined },
+        [voterWeighted2]:  { weight: 2,  outcome: undefined },
+        [voterWeighted3]:  { weight: 3,  outcome: undefined },
+        [voterWeighted10]: { weight: 10, outcome: undefined },
+        [voterWeighted12]: { weight: 12, outcome: undefined },
+        [voterWeighted13]: { weight: 13, outcome: undefined },
+      }
+
+      itDoesNotSetAWinningOutcome(votes)
+      itConsidersVotersAsLosers(votes, [ voterWeighted1, voterWeighted2, voterWeighted3, voterWeighted10, voterWeighted12, voterWeighted13 ])
```
Comment (izqui) [inline test/voting/crvoting-integration.js]: How about `Object.keys(votes)` to include all of them?

File: test/voting/crvoting-integration.js [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+          itConsidersVotersAsWinners(votes, expectedWinners)
+
+          const expectedLosers = [voterWeighted1, voterWeighted2, voterWeighted10, voterWeighted12, voterWeighted13]
+          itConsidersVotersAsLosers(votes, expectedLosers)
+        })
+      })
+
+      context('when there is tie', () => {
+        context('between the low and the high outcomes', () => {
+          const votes = {
```
Comment (izqui) [inline test/voting/crvoting-integration.js]: An interesting test could be to cast the votes in different order and ensure that the tally is still the same.

File: test/voting/crvoting-commit.js [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+        })
+      })
+
+      context('when the given voting ID is not valid', () => {
+        it('reverts', async () => {
+          await assertRevert(voting.commit(0, '0x', { from: voter }), 'CRV_VOTING_DOES_NOT_EXIST')
+        })
+      })
+
+      context('when the registry is not initialized', () => {
```
[the remark this answers is not in the document]
Comment (facuspagnuolo) [reply test/voting/crvoting-commit.js]: Oh, I noticed that I placed it within the wrong context, should be parallel to the `when the voting is initialized`. Nice catch!

File: test/voting/crvoting-create.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+      context('when the sender is not the owner', () => {
+        const from = someone
+
+        it('reverts', async () => {
+          await assertRevert(voting.create(1, 2, { from }), 'CRV_SENDER_NOT_OWNER')
+        })
+      })
+
+      context('when the registry is not initialized', () => {
```
[the remark this answers is not in the document]
Comment (facuspagnuolo) [reply test/voting/crvoting-create.js]: Yea, I noticed that I placed it within the wrong context, should be parallel to the "when the voting is initialized" context

**Pull request merged**